persistent-mysql/Database/Persist/MySQL.hs

```diff
@@ -1242,27 +1245,26 @@ mkBulkInsertQuery records fieldValues updates =
         , updateText
         ]
 
--- | Vendored from @persistent@.
-mkUpdateText :: PersistEntity record => Update record -> Text
-mkUpdateText x =
-  case updateUpdate x of
-    Assign -> n <> "=?"
-    Add -> T.concat [n, "=", n, "+?"]
-    Subtract -> T.concat [n, "=", n, "-?"]
-    Multiply -> T.concat [n, "=", n, "*?"]
-    Divide -> T.concat [n, "=", n, "/?"]
-    BackendSpecificUpdate up ->
-      error . T.unpack $ "BackendSpecificUpdate " <> up <> " not supported"
+putManySql :: EntityDef -> Int -> Text
+putManySql entityDef' numRecords
+  | numRecords > 0 = q
+  | otherwise = error "putManySql: numRecords MUST be greater than 0!"
```

Comment:
What do you think about making this a no-op if `== 0`? On the one hand inserting zero records could be a bug, on the other hand you might have code that dynamically changes the number of records inserted, and having zero be a no-op could be useful for that case. Thoughts?

Reply:
I agree with the no-op strategy. The users of `connPutManySql`: `putMany` and `defaultPutMany` already short circuit with a `return ()` when given 0 records. `putManySql` is never to be used unless there are records to be inserted (because it'd generate an invalid a sql query otherwise).